src/client/dfuse/pil4dfs/hook.c

```diff
@@ -152,6 +154,8 @@ read_map_file(char **buf, const char str_target[], char **pos)
 				break;
 			}
 		}
+		if (buffer_full)
+			free(*buf);
 		if (max_read_size >= MAP_SIZE_LIMIT) {
```

Comment:
If we have this hard limit on the size of the file, why not just allocate that and avoid the complication.   Probably better for fragmentation to do one large malloc/free than several of differing sizes.

Reply:
Normally the size of "/proc/self/maps" should be small, less than 100 K bytes. I would expect that the adaptively increasing buffer size only will occur in very rare cases.

Reply:
@jolivier23 free() and re-allocation should be not likely triggered since the default buffer size 512K is quite large already. Normally, "/proc/self/maps" during application initialization phase is far smaller.

Reply:
I was thinking the same, unexercised code is something we want to avoid.

Perhaps this would be a case for realloc() in a loop?